src/Common/src/Interop/Unix/System.Native/Interop.MAdvise.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+using size_t = System.UIntPtr;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [DllImport(Libraries.SystemNative, SetLastError = true)]
+        internal static extern int MAdvise(IntPtr addr, size_t length, MemoryAdvice advice);
```

Comment:
I'm still a bit torn on my size_t approach and considering going to ulong/uint64_t everywhere. 

Right now, it's actually declared as size_t in the native code and not uintptr_t, but there's a static assert to check that size_t is what we expect. 

UIntPtr on managed side is a pain (no arithmetic, comparison operators, etc.) On native side, if we used uint64_t, we'd have to do a few < SIZE_MAX checks (or have debt to do so when x86 comes online), but it would make the managed code easier to maintain I think.

I guess there are the following options:

(a) Managed=UIntPtr, Native=size_t, static assert that sizeof(size_t) == sizeof(uintptr_t)
(b) Managed=UIntPtr, Native=uintptr_t, static assert that sizeof(size_t) == sizeof(uintptr_t)
(c) Managed=ulong/Native=uint64_t, native fails calls if it sees size > SIZE_MAX

(a) is what I have now, but (c) is where I'm leaning. 

What do you think?